rp2040-hal/src/usb.rs

```diff
@@ -629,7 +630,7 @@ impl UsbBusTrait for UsbBus {
                 #[cfg(not(feature = "rp2040-e5"))]
                 return PollResult::Reset;
             } else if buff_status == 0 && sie_status.setup_rec().bit_is_clear() {
-                if sie_status.suspended().bit_is_set() {
+                if sie_status.suspended().bit_is_set() || ints.dev_suspend().bit_is_set() {
                     inner.ctrl_reg.sie_status.write(|w| w.suspended().set_bit());
                     return PollResult::Suspend;
```

Comment:
I wonder if it's correct to return PollResult::Suspend if sie_status.suspended() is not set.
Clearing the interrupt flag when only ints.dev_suspend is set looks like a valid fix, good catch!
Out of curiosity: what's the USB host you use when this issue occurs?

Reply:
I agree that this is probably not the correct way to fix this. Still discussing in matrix room.

The issue repros on a USB 2.0 hub, or direct to a USB 2.0 port on host, or USB 3.0 port on host.

Reply:
What kind of host? It might be operating system or even driver dependent.

Reply:
Linux fractal 5.15.70 #1-NixOS SMP Fri Sep 23 12:15:52 UTC 2022 x86_64 GNU/Linux